### PR TITLE
Watchdog: Allow sending notifications via Slack webhooks

### PR DESCRIPTION
**Pull request by jacobbaungard**
With this commit we allow watchdog notifications to be sent to a Slack
channel via a Slack webhook (https://api.slack.com/messaging/webhooks).

Notifications are sent to all defined methods. That if both watchdog
e-mail and slack webhook is defined, notifications are sent to both. If
only one of them are defined, only one notification method will be used.

In order to do so, a bunch of if statements were moved to the
`mail_error` function.

### CODE REVIEW

File: data/Dockerfiles/watchdog/watchdog.sh [hunk 1]
```diff
@@ -90,38 +90,46 @@ function mail_error() {
...
+        --hello-host=${MAILCOW_HOSTNAME} \
+        --ipv4
+        #--server="${RCPT_MX}"
+      log_msg "Sent notification email to ${rcpt}"
+    done
+  fi
+  # send slack msg, too
+  if [ ! -z "$WATCHDOG_SLACK_URL" ]; then
+    content="\"text\": \"${SUBJECT}\",\"blocks\": [{\"type\": \"section\",\"text\":{\"type\": \"mrkdwn\",\"text\":\"*${SUBJECT}*\"}},{\"type\": \"section\",\"text\":{\"type\": \"mrkdwn\",\"text\":\"${BODY}\"}}]"
+    curl -X POST --data-urlencode "payload={\"username\": \"mailcow-bot\", $content}" $WATCHDOG_SLACK_URL
```
Comment (ntimo) [inline data/Dockerfiles/watchdog/watchdog.sh]: I haven't used Slack that much, but does this mailcow-bot user actually have to exist inside of your Slack organisation? And should it be a hardcoded value?

Comment (jacobbaungard) [reply data/Dockerfiles/watchdog/watchdog.sh]: Thanks, good point. Turns out it is not needed to set the username at all, as it is set when the incoming webhook is created. I've amended the commit.

**Comment by jacobbaungard** [comment]
Any thoughts whether this feature is something you'd like to include?

**Comment by andryyy** [comment]
Hi,
Can we leave this open a bit longer? I can try to use this as base for a more general HTTP notification config. Is that fine for you? :)

**Comment by jacobbaungard** [comment]
Sure sounds good. Happy to assist in building something more general as well, just let me know :-)

**Comment by jacobbaungard** [comment]
@andryyy anything I can do to help? If you detail roughly what you are thinking I can try to implement something.

**Comment by GrantBirki** [comment]
👍 This feature would be awesome. I am currently running into an edge case where I believe outgoing SMTP is not possible (Azure VM) and this would be an excellent workaround. https://github.com/mailcow/mailcow-dockerized/issues/4197

**Comment by jacobbaungard** [comment]
Closing as https://github.com/mailcow/mailcow-dockerized/pull/4968 is merged.

**Pull request closed without merging**